### ISSUE TITLE
[BUG] http-proxy-middleware@1.1.1 : "Cannot read property 'host' of undefined" if target not set and router returns unresolvable URL.

### ISSUE THREAD

**Issue by dtoch** [chimurai/http-proxy-middleware#521]
### Is this a bug report?

YES

### Steps to reproduce

- package.json : 

```
{
  "name": "testbug",
  "version": "0.0.0",
  "dependencies": {
    "express": "4.17.1",
    "http-proxy-middleware": "1.1.1"
  }
}
```

- test.js

```
const express = require('express')

const app = express()

const { createProxyMiddleware  } = require('http-proxy-middleware');

const options = {
    router: function (req) {
       return "http://INVALID";
    }
};

app.use('/test', createProxyMiddleware (options));

app.listen(3000, () => {
  console.log(`Example app listening at http://localhost:3000`)
})
```

Then : 
- npm install
- node test.js
- curl http://localhost:3000/test

Result : following runtime error is logged :  

```
TypeError: Cannot read property 'host' of undefined
    at ProxyServer.HttpProxyMiddleware.logError (C:\Users\dtoch\work\projects\pga\pga-iaf\iaf-hil-common-server-node\node_modules\.pnpm\http-proxy-middleware@1.1.1_debug@4.3.1\node_modules\http-proxy-middleware\dist\http-proxy-middleware.js:141:87)
    at ProxyServer.emit (C:\Users\dtoch\work\projects\pga\pga-iaf\iaf-hil-common-server-node\node_modules\.pnpm\eventemitter3@4.0.7\node_modules\eventemitter3\index.js:210:27)
    at ClientRequest.proxyError (C:\Users\dtoch\work\projects\pga\pga-iaf\iaf-hil-common-server-node\node_modules\.pnpm\http-proxy@1.18.1_debug@4.3.1\node_modules\http-proxy\lib\http-proxy\passes\web-incoming.js:165:18)
```

### Expected behavior

Not a TypeError runtime error should be generated but instead a normal error just indicating that the backend URL is invalid.

### Actual behavior

TypeError: Cannot read property 'host' of undefined

### Workaround

According to https://github.com/chimurai/http-proxy-middleware/pull/512 part of http-proxy-middleware@1.1.1 it should not be needed to define target if using router. However to avoid above error, it is still needed to set it to something e.g. XXX.

### Setup

See above.

#### client info

- node 12.20.2
- npm 7.9.0

#### target server info

target not defined (only using router section in options).

**Comment by chimurai** [comment]
Thanks for reporting.
Think proposed solution in #242 might fix the issue

**Comment by chimurai** [comment]
Could you try and verify if `http-proxy-middleware@1.1.2-alpha.1` fixes the issue?

**Comment by dtoch** [comment]
I confirm the issue doesn't appear with http-proxy-middleware@1.1.2-alpha.1.

**Comment by chimurai** [comment]
Published `v1.1.2`

Implementation/fix in final  `v1.1.2` is different than the previous `http-proxy-middleware@1.1.2-alpha.1`.

Let me know if run into new issues.

**Comment by dtoch** [comment]
Thanks. I confirm the issue doesn't appear with the new stable release http-proxy-middleware@1.1.2.
